File: mobile/lib/models/teacher.dart
class Teacher {
  final String id;
  final String name;
  final String subject;
  final String email;
  final int classesCount;
  final int studentsCount;
  final String? avatarUrl;

  Teacher({
    required this.id,
    required this.name,
    required this.subject,
    required this.email,
    required this.classesCount,
    required this.studentsCount,
    this.avatarUrl,
  });
}

File: web/src/backend/models/Teacher.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface ITeacher extends Document {
  name: string;
  email: string;
  subject: string;
  avatar?: string;
  createdAt: Date;
  updatedAt: Date;
}

const TeacherSchema = new Schema<ITeacher>(
  {
    name: { type: String, required: true },
    email: { type: String, required: true, unique: true },
    subject: { type: String, required: true },
    avatar: { type: String },
  },
  { timestamps: true }
);

export default mongoose.models.Teacher || mongoose.model<ITeacher>('Teacher', TeacherSchema);
